### PR TITLE
Adding instrumentation to upload wizard selects (SCP-3678)

### PR DESCRIPTION
**Pull request by devonbush**
This ports over the recent instrumented select and common loading spinner to tis branch,

This also fixes an annoying bug where I would sign in after a session timeout but get stuck on the sign in page due to a script error caused by not escaping the quotes in a `"csrf required"` error

TO TEST:
1. Click through the new upload wizard, observe loading spinners
2. toggle the dropdowns on the cluster, expression and cooridnate label forms, and confirm mixpanel events are sent
3. allow session to timeout, then when prompted, sign back in.  Observe no errors in JS console (this might be tricky to reproduce because it depends on a specific csrf/token/session state

### CODE REVIEW

File: app/javascript/components/upload/CoordinateLabelFileForm.js [hunk 1]
```diff
@@ -33,12 +33,14 @@ export default function CoordinateLabelForm({
...
-            id={`coordCluster-${file._id}`}
-            value={associatedCluster}
-            placeholder="Select one"
-            onChange={val => updateCorrespondingClusters(file, val)}/>
+          <label className="labeled-select">Corresponding clusters / spatial data:
+            <Select options={associatedClusterFileOptions}
+              data-analytics-name="label-corresponding-cluster"
+              id={`coordCluster-${file._id}`}
+              value={associatedCluster}
+              placeholder="Select one"
```
Comment (eweitz) [inline app/javascript/components/upload/CoordinateLabelFileForm.js:41]: Our other menus with this phrase add an ellipsis, so for consistency...

```suggestion
              placeholder="Select one..."
```

(I personally prefer no ellipsis, but I prefer consistency more.)

File: app/javascript/components/upload/ExpressionFileForm.js [hunk 2]
```diff
@@ -105,15 +107,17 @@ function ExpressionFileInfoSelect({ label, propertyName, rawOptions, file, updat
...
-      value={selectedOption}
-      placeholder="Select one..."
-      onChange={val => {
-        const expInfo = {}
-        expInfo[propertyName] = val.value
-        updateFile(file._id, { expression_file_info: expInfo })
-      }}/>
+    <label className="labeled-select">{label}
+      <Select options={selectOptions}
+        data-analytics-name={`expression-${propertyName}-select`}
```
Comment (eweitz) [inline app/javascript/components/upload/ExpressionFileForm.js:112]: I (maybe naively) assume `propertyName` values are underscore-delimited as preferred by Ruby.  For consistency in Mixpanel, we should hyphen-delimit here.

```suggestion
        data-analytics-name={`expression-${_kebabCase(propertyName)}-select`}
```

(You'll need to import `_kebabCase`, of course.)

**Review by eweitz — APPROVED** [review]
Looks good!  I noted a few nits.

File: app/views/layouts/_notices.html.erb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
     if (<%= [alert, notice].compact.any? %>) {
-        showMessageModal("<%= notice.present? ? notice.html_safe : nil %>", "<%= alert.present? ? alert.html_safe : nil %>")
+        showMessageModal("<%= notice.present? ? notice.html_safe : nil %>", "<%= alert.present? ? alert.html_safe.gsub('"', '\'') : nil %>")
```
Comment (eweitz) [inline app/views/layouts/_notices.html.erb:3]: It'd be awesome if we move (even further) away from interpolated Ruby as part of this refactor.

**Review by bistline — APPROVED** [review]

**Pull request merged**